### ISSUE TITLE
Do you use DTO?

### ISSUE THREAD

**Issue by biapar** [workcontrolgit/TemplateKissAPI#2]
Do you use DTO?

**Comment by workcontrolgit** [comment]
Yes.  For me, I equate DTO to a view in the database.  In the traditional MVC, the DTO is great.  When working on REST API, I prefer CQRS pattern and implement using MediatR and AutoMapper.  It makes code cleaner and easier to test/maintain.

**Comment by biapar** [comment]
Automapper to map data class to model class returned by api. MediaR for? Which is a good panel to made CRUD operations on data as backend panel for a mobile app? I like Strapi.io but I like a similar thing for .NET Core so I've more control on it.

**Comment by biapar** [comment]
CQRS https://docs.microsoft.com/it-it/azure/architecture/patterns/cqrs
So you use MediaR to write data with commands and read by handlers. MediaR is like MessageBus into Xamarin. Then the commands after mediaR can be queued ti a queue and sent to database.

**Comment by biapar** [comment]
Maybe an update for this template.
